### PR TITLE
Repo Gardening: handle External Media extension

### PR DESCRIPTION
**Pull request by jeherve**
## Proposed changes:

This should allow automatically labeling PRs like #40382

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

No

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

This can be tested in a fork, by merging this branch to `trunk` and then opening a PR making changes to a file like `projects/plugins/jetpack/extensions/shared/external-media/media-button/media-ai-button.js`.

### CODE REVIEW

File: projects/github-actions/repo-gardening/src/tasks/add-labels/index.js [hunk 1]
```diff
@@ -218,6 +218,14 @@ async function getLabelsToAdd( octokit, owner, repo, number, isDraft, isRevert )
 			}
 		}
 
+		// External Media extension.
+		const externalMedia = file.match(
+			/^projects\/plugins\/jetpack\/extensions\/shared\/external-media\//
+		);
+		if ( externalMedia !== null ) {
```
Comment (tbradsha) [inline projects/github-actions/repo-gardening/src/tasks/add-labels/index.js:225]: Not a blocker (and I see it's done like this elsewhere), but I prefer simplifying the construct: 
```suggestion
		if ( externalMedia ) {
```

**Review by tbradsha — APPROVED** [review]
Approving, as this works as described.

More generally I do wonder if we have a label bloat problem; do we really need such specific labels?

**Comment by jeherve** [comment]
> More generally I do wonder if we have a label bloat problem; do we really need such specific labels?

That's a good question. They do prove useful for stats, reporting, and for the automations (like pinging in Slack or adding to project boards) so they are useful, but I do think that for some of our products, they're not useful (i.e. the "[Feature] Photon" and "[Package] Image CDN" could probably be merged into one if we updated our logic). 

I'll see about starting a discussion about this. 👍

File: projects/github-actions/repo-gardening/src/tasks/add-labels/index.js (hunk 1 above)
Comment (jeherve) [reply projects/github-actions/repo-gardening/src/tasks/add-labels/index.js:225]: I'll see about updating them all at once in a future PR!

**Pull request merged**